brian2/core/network.py

```diff
@@ -853,27 +859,31 @@ def run(self, duration, report=None, report_period=10*second,
                 timestep, _, _ = self._clock_variables[clock]
                 running = timestep < clock._i_end
 
+        end_time = time.time()
         if self._stopped or Network._globally_stopped:
             self.t_ = clock.t_
         else:
             self.t_ = float(t_end)
 
-        device._last_run_time = time.time()-start_time
+        device._last_run_time = end_time-start_time
         if duration>0:
             device._last_run_completed_fraction = (self.t-t_start)/duration
         else:
             device._last_run_completed_fraction = 1.0
 
         if report is not None:
-            report_callback((current-start)*second, 1.0, duration)
+            report_callback((end_time-start_time)*second, 1.0, t_start, duration)
         self.after_run()
 
+        logger.debug(("Finished simulating network '%s' "
+                      "(took %.2fs)") % (self.name, end_time-start_time),
+                     'run')
         # Store profiling info (or erase old info to avoid confusion)
         if profile:
             self._profiling_info = [(name, t*second)
                                     for name, t in profiling_info.iteritems()]
             # Dump a profiling summary to the log
-            logger.debug('\n' + str(profiling_summary(self)))
+            logger.diagnostic('\n' + str(profiling_summary(self)))
```

Comment:
Yeah, I wouldn't mind. Another potential candidate is the information about which `SpikeQueue` is used (maybe only in case it is the slower Python queue?).